File: 70.js
// Fibonacci solution
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let curr = 0;
    let last = 1;

    while (n >= 0) {
        const temp = last;
        last = curr;
        curr = curr + temp;
        n--;
    }

    return curr;
};

File: 70.rs
// Fibonacccy solution
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut curr = 1;
        let mut prev = 0;

        for n in 0..n {
            (curr, prev) = (curr + prev, curr);
        }

        return curr;
    }
}

